libraries/rib-base/src/main/java/com/badoo/ribs/core/view/ViewFactory.kt

```diff
@@ -1,4 +1,4 @@
 package com.badoo.ribs.core.view
 
-interface ViewFactory<Dependency, View>: (Dependency) -> (RibView) -> View
+interface ViewFactory<Dependency, View : RibView> : (Dependency) -> NodeViewFactory<View>
```

Comment:
Maybe this is the actual `ViewFactory`, and the "outer" one resolving `Dependency` is a... `Builder`?